webapp/tests/conftest.py

```diff
@@ -36,21 +33,52 @@ def test_request_context(app):
         yield req
 
 
+CURRENT_USER_IDNR = "0123456789"
+
+
+@pytest.fixture(autouse=True)
+def testing_current_user(monkeypatch):
+    monkeypatch.setattr("app.forms.session_data.current_user", MagicMock(idnr_hashed=CURRENT_USER_IDNR))
+
+
+@pytest.fixture(autouse=True)
+def testing_database(monkeypatch):
+    fakeredis_connection = fakeredis.FakeStrictRedis()
+    monkeypatch.setattr("app.data_access.form_data_controller.FormDataController._redis_connection",
+                        fakeredis_connection)
+    yield
+    fakeredis_connection.flushall()
+
+
 @pytest.fixture
 def new_test_request_context(app):
     @contextmanager
     def _new_test_request_context(method='GET', form_data=None, stored_data=None, session_identifier='form_data'):
+        from app.forms.session_data import serialize_session_data
+        from app.data_access.form_data_controller import FormDataController
         with app.test_request_context() as req:
             req.request.method = method
             if stored_data:
-                req.session = SecureCookieSession({session_identifier: create_session_form_data(stored_data)})
-            if form_data:
-                req.request.data = ImmutableMultiDict(form_data)
-                req.request.form = ImmutableMultiDict(form_data)
+                FormDataController().save_to_redis(CURRENT_USER_IDNR + '_' + session_identifier,
+                                                   serialize_session_data(stored_data))
+            else:
+                FormDataController().save_to_redis(CURRENT_USER_IDNR + '_' + session_identifier,
+                                                   serialize_session_data(form_data))
             yield req
+
     return _new_test_request_context
 
 
+@pytest.fixture
+def test_request_context_with_person_b_disability(app):
+    from app.forms.session_data import serialize_session_data
+    from app.data_access.form_data_controller import FormDataController
+    with app.test_request_context(method="POST") as req:
+        FormDataController().save_to_redis(CURRENT_USER_IDNR + '_form_data',
+                                            serialize_session_data({'person_b_has_disability': 'yes'}))
+        yield req
```

Comment:
We could use the fixture from above. wdyt?

Reply:
Done